components/brave_wallet/common/eth_request_helper.cc

```diff
@@ -677,19 +677,84 @@ bool ParseEthSendRawTransactionParams(const std::string& json,
   return true;
 }
 
-bool ParseEthSubscribeParams(const std::string& json, std::string* event_type) {
+bool ParseEthSubscribeParams(const std::string& json,
+                             std::string* event_type,
+                             std::vector<std::string>* address,
+                             std::string* from_block,
+                             std::string* to_block,
+                             std::string* block_hash,
+                             std::vector<std::string>* topics) {
   if (!event_type)
     return false;
 
   auto list = GetParamsList(json);
-  if (!list || list->size() != 1)
+  if (!list)
     return false;
 
-  const std::string* event_type_str = (*list)[0].GetIfString();
-  if (!event_type_str)
-    return false;
+  const auto get_string_val = [](const std::string& name,
+                                 const base::Value::Dict* filtering_params,
+                                 std::string* vout) {
+    if (vout == nullptr)
+      return;
+
+    const base::Value* vvalue_str = filtering_params->Find(name);
+
+    if (vvalue_str == nullptr)
+      return;
+
+    if (const auto* value_str = (*vvalue_str).GetIfString()) {
+      *vout = *value_str;
+    }
+  };
+
+  const auto get_array = [](const std::string& name,
+                            const base::Value::Dict* filtering_params,
+                            std::vector<std::string>* arr_out) {
+    if (arr_out == nullptr)
+      return;
+
+    const base::Value* vvalue = filtering_params->Find(name);
+
+    if (vvalue == nullptr)
+      return;
+
+    if (vvalue->is_string()) {
+      arr_out->emplace_back(*vvalue->GetIfString());
+      return;
+    }
+
+    const auto* vvalue_list = vvalue->GetIfList();
+
+    if (vvalue_list == nullptr)
+      return;
+
+    for (const auto& topics_list_item : (*vvalue_list)) {
+      if (const auto* topics_list_item_str = topics_list_item.GetIfString()) {
+        arr_out->emplace_back(*topics_list_item_str);
+      }
+    }
```

Comment:
I think topics filter for get_ethLogs and eth_newFilter has same format
https://ethereum.org/ru/developers/docs/apis/json-rpc/#eth_newfilter
which means [A, [B, C]] and [A, B, C] are different filters, but this parser flattens such array. Nulls are also supposed to be supported.

Reply:
Well it seems to me logging params should be transparent to browser, i.e. we should get filter params as base::Value from renderer and pass it to rpc endpoint as is.
So interface should look like this
```
void JsonRpcService::EthGetLogs(const std::string& chain_id,
                                const base::Value& params,
                                EthGetLogsCallback callback)
```

Reply:
fixed